PORTAL/jobs/__main__.py

```diff
@@ -245,18 +286,35 @@ def cmd_wait(jobs, reqid=None):
         pass
 
 
-def cmd_upload(jobs, reqid, *, author=None, clean=True, push=True):
+def cmd_upload(
+        jobs: Jobs,
+        reqid: RequestID,
+        *,
+        author: Optional[str] = None,
+        clean: bool = True,
+        push: bool = True
+) -> None:
     job = jobs.get(reqid)
-    job.upload_result(author, clean=clean, push=push)
+    if job:
+        job.upload_result(author, clean=clean, push=push)
+    else:
+        raise RuntimeError(f"Couldn't get job for {reqid}")
 
 
-def cmd_compare(jobs, res1, others, *, meanonly=False, pyston=False):
+def cmd_compare(
+        jobs: Jobs,
+        res1: ToRequestIDType,
+        others: List[Any],
```

Comment:
These have the same type as `res1`.  (So `Sequence[...]`.)

Reply:
Hmm...  Type checker says that it's adding `PyperfResultsFile` objects to it below (in `others.extend(matched)`)...

Reply:
As with `res1` this is only ever reqid strings or filenames., so:

```suggestion
        others: List[str],
```

If mypy is fussing about variable reuse then I guess we'll have to introduce a new variable like you did with `res1_matched`.  "others_matched"?